consensus/state_sync/server.go

```diff
@@ -0,0 +1,125 @@
+package state_sync
+
+import (
+	"fmt"
+
+	typesCons "github.com/pokt-network/pocket/consensus/types"
+	"github.com/pokt-network/pocket/shared/codec"
+	coreTypes "github.com/pokt-network/pocket/shared/core/types"
+	cryptoPocket "github.com/pokt-network/pocket/shared/crypto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+// This module is responsible for handling requests and business logic that advertises and shares
+// local state metadata with other peers synching to the latest block.
+type StateSyncServerModule interface {
+	// Advertise (send) the local state sync metadata to the requesting peer
+	HandleStateSyncMetadataRequest(*typesCons.StateSyncMetadataRequest) error
+
+	// Send the block being requested by the peer
+	HandleGetBlockRequest(*typesCons.GetBlockRequest) error
+}
+
+func (m *stateSync) HandleStateSyncMetadataRequest(metadataReq *typesCons.StateSyncMetadataRequest) error {
+	serverNodePeerId, err := m.GetBus().GetConsensusModule().GetCurrentNodeAddressFromNodeId()
+	if err != nil {
+		return err
+	}
+
+	clientPeerId := metadataReq.PeerId
+	m.nodeLog(fmt.Sprintf("%s received State Sync MetaData Req from: %s", serverNodePeerId, clientPeerId))
+
+	minHeight, maxHeight := m.aggregateMetaResults()
+
+	stateSyncMessage := typesCons.StateSyncMessage{
+		MsgType: typesCons.StateSyncMessageType_STATE_SYNC_METADATA_RESPONSE,
+		Message: &typesCons.StateSyncMessage_MetadataRes{
+			MetadataRes: &typesCons.StateSyncMetadataResponse{
+				PeerId:    serverNodePeerId,
+				MinHeight: minHeight,
+				MaxHeight: maxHeight,
+			},
+		},
+	}
+
+	anyMsg, err := anypb.New(&stateSyncMessage)
+	if err != nil {
+		return err
+	}
+
+	return m.sendToPeer(anyMsg, cryptoPocket.AddressFromString(clientPeerId))
+}
+
+func (m *stateSync) HandleGetBlockRequest(blockReq *typesCons.GetBlockRequest) error {
+	consensusMod := m.GetBus().GetConsensusModule()
+	serverNodePeerId, err := consensusMod.GetCurrentNodeAddressFromNodeId()
+	if err != nil {
+		return err
+	}
+
+	clientPeerId := blockReq.PeerId
+
+	m.nodeLog(fmt.Sprintf("%s received State Sync Get Block Req from: %s", serverNodePeerId, clientPeerId))
+
+	// IMPROVE: Consider checking the hishest block from persistance, rather than the consensus module
+	// check the max block height, if higher height is requested, return error
+	if consensusMod.CurrentHeight() < blockReq.Height {
+		return fmt.Errorf(" requested block height is higher than node's block height ")
```

Comment:
We should update the error to contain the request height and the max available height

```suggestion
		return fmt.Errorf("requested block height is higher than node's block height ")
```